Mathlib/Analysis/SpecialFunctions/Log/Deriv.lean

```diff
@@ -0,0 +1,340 @@
+    ContDiffOn ℝ n (fun x => log (f x)) s := fun x hx => (hf x hx).log (hs x hx)
+#align cont_diff_on.log ContDiffOn.log
+
+theorem ContDiff.log {n} (hf : ContDiff ℝ n f) (h : ∀ x, f x ≠ 0) :
+    ContDiff ℝ n fun x => log (f x) :=
+  contDiff_iff_contDiffAt.2 fun x => hf.contDiffAt.log (h x)
+#align cont_diff.log ContDiff.log
+
+theorem DifferentiableOn.log (hf : DifferentiableOn ℝ f s) (hx : ∀ x ∈ s, f x ≠ 0) :
+    DifferentiableOn ℝ (fun x => log (f x)) s := fun x h => (hf x h).log (hx x h)
+#align differentiable_on.log DifferentiableOn.log
+
+@[simp]
+theorem Differentiable.log (hf : Differentiable ℝ f) (hx : ∀ x, f x ≠ 0) :
+    Differentiable ℝ fun x => log (f x) := fun x => (hf x).log (hx x)
+#align differentiable.log Differentiable.log
+
+theorem fderivWithin.log (hf : DifferentiableWithinAt ℝ f s x) (hx : f x ≠ 0)
+    (hxs : UniqueDiffWithinAt ℝ s x) :
+    fderivWithin ℝ (fun x => log (f x)) s x = (f x)⁻¹ • fderivWithin ℝ f s x :=
+  (hf.hasFDerivWithinAt.log hx).fderivWithin hxs
+#align fderiv_within.log fderivWithin.log
+
+@[simp]
+theorem fderiv.log (hf : DifferentiableAt ℝ f x) (hx : f x ≠ 0) :
+    fderiv ℝ (fun x => log (f x)) x = (f x)⁻¹ • fderiv ℝ f x :=
+  (hf.hasFDerivAt.log hx).fderiv
+#align fderiv.log fderiv.log
+
+end fderiv
+
+end LogDifferentiable
+
+namespace Real
+
+/-- The function `x * log (1 + t / x)` tends to `t` at `+∞`. -/
+theorem tendsto_mul_log_one_plus_div_atTop (t : ℝ) :
+    Tendsto (fun x => x * log (1 + t / x)) atTop (𝓝 t) := by
+  have h₁ : Tendsto (fun h => h⁻¹ * log (1 + t * h)) (𝓝[≠] 0) (𝓝 t) := by
+    simpa [hasDerivAt_iff_tendsto_slope, slope_fun_def] using
+      (((hasDerivAt_id (0 : ℝ)).const_mul t).const_add 1).log (by simp)
+  have h₂ : Tendsto (fun x : ℝ => x⁻¹) atTop (𝓝[≠] 0) :=
+    tendsto_inv_atTop_zero'.mono_right (nhdsWithin_mono _ fun x hx => (Set.mem_Ioi.mp hx).ne')
+  simpa only [(· ∘ ·), inv_inv] using h₁.comp h₂
+#align real.tendsto_mul_log_one_plus_div_at_top Real.tendsto_mul_log_one_plus_div_atTop
+
+open scoped BigOperators
+
+/-- A crude lemma estimating the difference between `log (1-x)` and its Taylor series at `0`,
+where the main point of the bound is that it tends to `0`. The goal is to deduce the series
+expansion of the logarithm, in `has_sum_pow_div_log_of_abs_lt_1`.
+
+Porting note: TODO: use one of generic theorems about Taylor's series to prove this estimate.
+-/
+theorem abs_log_sub_add_sum_range_le {x : ℝ} (h : |x| < 1) (n : ℕ) :
+    |(∑ i in range n, x ^ (i + 1) / (i + 1)) + log (1 - x)| ≤ |x| ^ (n + 1) / (1 - |x|) := by
+  /- For the proof, we show that the derivative of the function to be estimated is small,
+    and then apply the mean value inequality. -/
+  let F : ℝ → ℝ := fun x => (∑ i in range n, x ^ (i + 1) / (i + 1)) + log (1 - x)
+  let F' : ℝ → ℝ := fun x ↦ -x ^ n / (1 - x)
+  -- Porting note: In `mathlib3`, the proof used `deriv`/`DifferentiableAt`. `simp` failed to
+  -- compute `deriv`, so I changed the proof to use `HasDerivAt` instead
```

Comment:
Perhaps add a link to https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/!4.234653.20Trigonometric.2EDeriv/near/363605590, where a related failure of `simp` to compute derivatives is being discussed.

Reply:
Oh, too late, I see, that was the previous PR already. :-)